test/functional/feature_staking_unlock_and_withdraw.py

```diff
@@ -87,9 +88,11 @@ def run_test(self):
         assert_equal(events[2].value['event_id'], 'StakeUnlocked')
 
         ledger = list(client.get_staking_ledger())
+
         assert_equal(len(ledger),2)
 
-        time.sleep(500)
+        while client.current_era() < client.withdrawal_era(alice):
+            time.sleep(1)
```

Comment:
I think there should be a hard timeout as well in case there is a bug somewhere that causes `withdrawal_era` to never be reached.

Reply:
I agree. I will add it.